### PR TITLE
[Refactor] Refactor launch function and RemoteOptions

### PR DESCRIPTION
**Pull request by jimchou-dev**
## Purpose

* Refactor launch function
* Refactor RemoteOptions to replace GroupCallOptions and TeamsMeetingOptions
* Introduce CallCompositeLocator enum, may be renamed in separate PR
* Documentation/Changelog update will be in separate PR

## Does this introduce a breaking change?

```
[ x ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ x ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* All unit test and UI test are passing, app is functional

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/CallConfiguration.swift [hunk 1]
```diff
@@ -7,32 +7,29 @@ import Foundation
...
-        self.diagnosticConfig = DiagnosticConfig()
-    }
-
-    init(credential: CommunicationTokenCredential,
-         meetingLink: String,
+    init(locator: CallCompositeLocator,
+         credential: CommunicationTokenCredential,
          displayName: String?) {
+        switch locator {
+        case let .groupCall(groupId: groupId):
```
[the remark this answers is not in the document]
Comment (jimchou-dev) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/CallConfiguration.swift:21]: Just check with ARB for iOS, they seem to be okay with this enum as it's unlikely that customer will need to be performing a switch case on this enum. They should be providing this to us, and not the other way around.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/RemoteOptions.swift [hunk 2]
```diff
@@ -6,41 +6,48 @@
 import Foundation
 import AzureCommunicationCommon
 
-/// Options for joining a group call.
-public struct GroupCallOptions {
+/// CallComposite Locator for locating call destination
+public enum CallCompositeLocator {
+    /// Group Capp with UUID groupId
```
Comment (kevinyulu) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/RemoteOptions.swift:11]: typo: `call`?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/CallCompositeOptions/RemoteOptionsTests.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+//
+
+import Foundation
+import XCTest
+import AzureCommunicationCommon
+@testable import AzureCommunicationUICalling
+
+class RemoteOptionsTests: XCTestCase {
+    func test_remoteOptions_init_groupCall_when_parametersAreValid_then_returnRemoteOptionsObject() {
+        let sampleToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjMyNTAzNjgwMDAwfQ.9i7FNNHHJT8cOzo-yrAUJyBSfJ-tPPk2emcHavOEpWc"
```
Comment (kevinyulu) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/CallCompositeOptions/RemoteOptionsTests.swift:13]: just curious, would this sampleToken be expired at some point?

**Review by kevinyulu — DISMISSED** [review]
tested with iPhone XR, looks good.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/RemoteOptions.swift (hunk 2 above)
Comment (jimchou-dev) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/RemoteOptions.swift:11]: Good catch!

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/CallCompositeOptions/RemoteOptionsTests.swift (hunk 3 above)
Comment (jimchou-dev) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/CallCompositeOptions/RemoteOptionsTests.swift:13]: This is just a partial token moved over from the GroupCallOptions unit test

**Review by kevinyulu — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/CallConfiguration.swift [hunk 4]
```diff
@@ -7,32 +7,29 @@ import Foundation
...
+            print(groupId)
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/CallConfiguration.swift:22]: should we remove print?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/RemoteOptions.swift [hunk 5]
```diff
@@ -6,41 +6,48 @@
...
-
     /// The display name of the local participant when joining the call.
     ///
     /// The limit for string length is 256.
     public let displayName: String?
 
-    /// Create an instance of a GroupCallOptions with options.
+    /// Create an instance of a RemoteOptions with options.
     /// - Parameters:
+    ///   - for: The CallCompositeLocator type with unique identifier for joining a specific call.
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/RemoteOptions.swift:32]: I think the parameter's name should be `locator`

**Review by vhuseinova-msft — APPROVED** [review]

**Review by palatter — APPROVED** [review]

**Review by kevinyulu — APPROVED** [review]

**Pull request merged**